controller/eks-cluster-config-handler.go

```diff
@@ -495,6 +455,56 @@ func (h *Handler) create(config *v13.EKSClusterConfig, sess *session.Session, ek
 	return h.eksCC.UpdateStatus(config)
 }
 
+func (h *Handler) generateAndSetNetworking(svc *cloudformation.CloudFormation, config *v13.EKSClusterConfig) (*v13.EKSClusterConfig, error) {
+	if len(config.Status.Subnets) != 0 {
+		// networking fields have already been set
+		return config, nil
+	}
+
+	if len(config.Spec.Subnets) != 0 {
+		logrus.Infof("VPC info provided, skipping vpc/subnet/securitygroup creation")
+		config = config.DeepCopy()
+		// copy networking fields to status
+		config.Status.Subnets = config.Spec.Subnets
+		config.Status.SecurityGroups = config.Spec.SecurityGroups
+		config.Status.NetworkFieldsSource = "provided"
+		var err error
+		config, err = h.eksCC.UpdateStatus(config)
+		if err != nil {
+			return config, err
+		}
+	} else {
+		logrus.Infof("Bringing up vpc")
+
+		stack, err := createStack(svc, getVPCStackName(config.Spec.DisplayName), config.Spec.DisplayName, templates.VpcTemplate, []string{},
+			[]*cloudformation.Parameter{})
+		if err != nil {
+			return config, fmt.Errorf("error creating stack with VPC template: %v", err)
+		}
+
+		virtualNetworkString := getParameterValueFromOutput("VpcId", stack.Stacks[0].Outputs)
+		securityGroupsString := getParameterValueFromOutput("SecurityGroups", stack.Stacks[0].Outputs)
+		subnetIdsString := getParameterValueFromOutput("SubnetIds", stack.Stacks[0].Outputs)
+
+		if securityGroupsString == "" || subnetIdsString == "" {
+			return config, fmt.Errorf("no security groups or subnet ids were returned")
+		}
+
+		config = config.DeepCopy()
+		// copy generated field to status
+		config.Status.VirtualNetwork = virtualNetworkString
+		config.Status.SecurityGroups = strings.Split(securityGroupsString, ",")
+		config.Status.Subnets = strings.Split(subnetIdsString, ",")
+		config.Status.NetworkFieldsSource = "generated"
+		config, err = h.eksCC.UpdateStatus(config)
+		if err != nil {
+			return config, err
+		}
+	}
+
+	return config, nil
```

Comment:
Once we remove the else (as per my previous comment), we could just replace this with `return h.eksCC.UpdateStatus(config)`

Reply:
We cannot remove the else and move the updateStatus to the bottom because if the first condition is met the fields will be overwritten by the logic currently contained in the else. We can either return early with update in the first block or move the overlapping update from both to the bottom.

Reply:
Went with keeping the else and returning the overlapping update status at the bottom.